enhancements/olm/single-ownnamespace-enhancement-proposal.md

```diff
@@ -0,0 +1,418 @@
+
+### User Stories
+
+#### Story 1: Legacy Operator Migration
+As a cluster administrator migrating from OLM v0 to OLM v1, I want to install operators that only support SingleNamespace or OwnNamespace install modes so that I can continue using existing operator content without requiring operator authors to modify their bundles.
+
+#### Story 2: Security-Conscious Deployments
+As a security-conscious administrator, I want to install operators that explicitly limit their scope to specific namespaces so that I can maintain strict RBAC boundaries and reduce the attack surface of operator deployments.
+
+#### Story 2: Operator Author Requirements
+As an operator developer, I have operators designed for namespace isolation that only support Single or OwnNamespace modes, and I want my customers to be able to deploy these operators in OpenShift with OLM v1 so that I can maintain the intended isolation boundaries while benefiting from OLM v1's simplified management.
+
+### Goals
+
+- Enable installation of registry+v1 bundles that only support SingleNamespace or OwnNamespace install modes
+- Provide a configuration mechanism through the ClusterExtension API to specify watch namespaces
+- Maintain backward compatibility with existing OLM v0 operator content
+- Generate appropriate RBAC resources scoped to the target namespaces
+- Support seamless migration path for customers moving from OLM v0 to OLM v1
+
+### Non-Goals
+
+- Re-introducing multi-tenancy features or supporting multiple installations of the same operator
+- Supporting MultiNamespace install mode (watching multiple namespaces)
+- Modifying the fundamental OLM v1 architecture or adding complex multi-tenancy logic
+- Supporting install mode switching after initial installation as a first-class feature
+- Enabling this feature for bundle formats other than registry+v1
+
+## Proposal
+
+Update the OLMv1 operator-controller to:
+1. Validate bundle compatibility with the requested install mode during resolution
+2. Parse the `watchNamespace` configuration from ClusterExtension.spec.config.inline
+3. Determine the install mode based on the relationship between install namespace and watch namespace
+4. Generate appropriate RBAC resources (Roles/RoleBindings vs ClusterRoles/ClusterRoleBindings) based on the determined install mode
+5. Configure operator deployments with correct environment variables and RBAC for the target watch namespace
+
+Add support for namespace-scoped operation by ensuring:
+1. Bundle validation confirms the operator supports the requested install mode
+2. RBAC resources are properly scoped to the watch namespace for Single/OwnNamespace modes
+3. Operator deployment environment variables are configured for the specified watch namespace
+4. Installation fails gracefully with clear error messages when bundles don't support the requested mode
```

Comment:
5. Use reasonable default install mode when possible:
 - AllNamespaces by default (when available)
 - OwnNamespace by default (when no AllNamespaces and available)